semgrep/semgrep/target_manager.py

```diff
@@ -237,23 +238,31 @@ def filtered_files(self, lang: Language) -> Set[Path]:
         if lang in self._filtered_targets:
             return self._filtered_targets[lang]
 
-        targets = self.resolve_targets(self._targets)
-        explicit_files, directories = partition_set(lambda p: not p.is_dir(), targets)
+        targets = self.resolve_targets(self.targets)
+
+        files, directories = partition_set(lambda p: not p.is_dir(), targets)
+
+        # Error on non-existent files
+        explicit_files, nonexistent_files = partition_set(lambda p: p.is_file(), files)
```

Comment:
You might need to handle symlinked files as well?

Reply:
It looks like is_file() should cover regular files and symlinks (and I just tested with symlinks and got the behavior I expected- but maybe that is not the desired behavior): https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file 
Or do you mean that symlinked files should be handled differently instead of lumped in with regular files?

Reply:
Oh hmm so it does. Not sure why tests are failing then

Reply:
At least one of them is because we added the parameter to TargetManager but the test file still thinks it only takes 4 arguments... I'm working on it